cmd/keytransparency-sequencer/main.go

```diff
@@ -202,15 +209,9 @@ func main() {
 		election.NewTracker(electionFactory, 1*time.Hour, prometheus.MetricFactory{}),
 	)
 
-	cctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	sequencer.PeriodicallyRun(cctx, time.Tick(*refresh), func(ctx context.Context) {
+	sequencer.PeriodicallyRun(ctx, time.Tick(*refresh), func(ctx context.Context) {
 		if err := signer.RunBatchForAllDirectories(ctx); err != nil {
 			glog.Errorf("PeriodicallyRun(RunBatchForAllDirectories): %v", err)
 		}
 	})
-
-	// Shutdown.
-	httpServer.Shutdown(cctx)
```

Comment:
How is the server shut down now?

Reply:
Reading the [documentation](https://golang.org/pkg/net/http/#Server.Shutdown) closely, `Shutdown` looks like it's intended to be called from a separate thread than the serving thread.  
If there was a way to trigger a shutdown, we would call it there, but at the moment this open source main function has SIGINT as the only shutdown signal.